### PR TITLE
[NNlibAMDGPUExt] Load MIOpen module only if it is available

### PR DESCRIPTION
**Pull request by pxl-th**
Since MIOpen artifact is only available on Julia 1.9 it is not included in AMDGPU by default.
Therefore we need to load it only when it is available.

Otherwise, it causes crashes for users that do not have MIOpen installed.
For example in AMDGPU tests, where we compare CPU with AMDGPU implementations and don't rely on NNlib AMDGPU support.

- Migrate to separate `Project.toml` for tests.

### PR Checklist

- [ ] Tests are added
- [ ] Documentation, if applicable

### CODE REVIEW

**Comment by ToucheSir** [comment]
1.6 CUDA failure is probably spurious, but the AMDGPU one does not seem to be.

**Comment by pxl-th** [comment]
Segfault in CI, because it uses global ROCm installation which is of incompatible version.
I've switched to using artifacts, but since MIOpen is not installed it skips tests.

**Comment by ToucheSir** [comment]
Can we install the MIOpen_jll on CI?

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -76,18 +76,23 @@ steps:
         push!(conf[\"targets\"][\"test\"], \"AMDGPU\");
         open(io -> TOML.print(io, conf), \"Project.toml\", \"w\");
         """
+      - julia --project=. -e """
+        using Pkg;
+        Pkg.develop(\"AMDGPU\");
```
Comment (ToucheSir) [inline .buildkite/pipeline.yml:81]: Another reason to switch to test/Project.toml sooner than later?

Comment (ToucheSir) [reply .buildkite/pipeline.yml:81]: Or now, thanks!

**Comment by pxl-th** [comment]
AMDGPU CI is probably down...

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -76,18 +76,23 @@ steps:
         push!(conf[\"targets\"][\"test\"], \"AMDGPU\");
```
Comment (ToucheSir) [inline .buildkite/pipeline.yml:76]: CI is breaking on `[targets]` no longer existing in Project.toml.

**Comment by pxl-th** [comment]
Should we merge it? Even though AMDGPU CI is down, should be safe to do so.

**Review by ToucheSir — APPROVED** [review]
Sounds good. I thought the CUDA jobs were also failing but must've not checked recently enough.

**Pull request merged**